src/ui/components/Button/Button.scss

```diff
@@ -1,20 +1,51 @@
 @import '~core/css/inc/mixins';
-@import '~core/css/inc/vars';
+@import '~ui/css/vars';
 
-.Button,
-.Button:link {
-  @include button();
+.Button--styleName-neutral {
+  @include button(
+    $background: $neutral-base-color,
+    $background-active: $neutral-active-color,
+    $background-hover: $neutral-hover-color
+  );
 }
 
-.Button--light,
-.Button--light:link {
-  @include button(#fff, $link-color);
+.Button--styleName-light {
+  @include button(
+    $background: $light-base-color,
+    $background-active: $light-active-color,
+    $background-hover: $light-hover-color,
+    $color: $type-black
+  );
 }
 
-.Button,
-.Button:link {
-  &.Button--small {
-    border-radius: $border-radius-s;
-    font-size: 13px;
-  }
+.Button--styleName-action {
+  @include button(
+    $background: $action-base-color,
+    $background-active: $action-active-color,
+    $background-hover: $action-hover-color
+  );
+}
+
+.Button--styleName-confirm {
```

Comment:
I think this could be simpler as `.Button--confirm` sticking to the modifier convention.